src/test/java/hudson/plugins/plot/CSVReaderTest.java

```diff
@@ -22,77 +22,77 @@
-            String[] nextLine;
-            // read each line of the CSV file and add to rawPlotData
-            int lineNum = 0;
-            while ((nextLine = csvreader.readNext()) != null) {
-                // for some reason csv reader returns an empty line sometimes.
-                if (nextLine.length == 1 && nextLine[0].length() == 0) {
-                    break;
-                }
+                // save the header line to use it for the plot labels.
+                String[] nextLine;
+                // read each line of the CSV file and add to rawPlotData
+                int lineNum = 0;
+                while ((nextLine = csvreader.readNext()) != null) {
+                    // for some reason csv reader returns an empty line sometimes.
+                    if (nextLine.length == 1 && nextLine[0].length() == 0) {
+                        break;
+                    }
 
-                if (COLUMNS[0] != nextLine.length) {
-                    StringBuilder msg = new StringBuilder();
-                    msg.append("column count is not equal ").append(nextLine.length);
-                    msg.append(" expected ").append(COLUMNS[0]).append(" at line ");
-                    msg.append(lineNum).append(" line: ").append("'");
-                    for (String s : nextLine) {
-                        msg.append("\"").append(s).append("\":").append(s.length()).append(",");
+                    if (COLUMNS[testfilenum] != nextLine.length) {
+                        StringBuilder msg = new StringBuilder();
+                        msg.append("column count is not equal ").append(nextLine.length);
+                        msg.append(" expected ").append(COLUMNS[testfilenum]).append(" at line ");
+                        msg.append(lineNum).append(" line: ").append("'");
+                        for (String s : nextLine) {
+                            msg.append("\"").append(s).append("\":").append(s.length()).append(",");
+                        }
+                        msg.append("' length ").append(nextLine.length);
+                        assertTrue(msg.toString(), COLUMNS[testfilenum] == nextLine.length);
                     }
-                    msg.append("' length ").append(nextLine.length);
-                    assertTrue(msg.toString(), COLUMNS[0] == nextLine.length);
-                }
-                ++lineNum;
-            }
-            assertTrue("Line count is not equal " + lineNum + " expected " + LINES[0], LINES[0] == lineNum);
-        } catch (IOException | InterruptedException e) {
-            assertFalse("Exception " + e, true);
-        } finally {
-            try {
-                if (csvreader != null) {
-                    csvreader.close();
+                    ++lineNum;
                 }
-            } catch (IOException e) {
+                assertTrue("Line count is not equal " + lineNum + " expected " + LINES[testfilenum], LINES[testfilenum] == lineNum);
```

Comment:
But if you can reformat the code according to `max line length = 100` rule, that would be awesome.

Reply:
@vgaidarji For testing on MacOS, you only need to make sure that a trailing space is added at the end of the last label *and* the last value in each row.  "echo" in MacOS or Linux probably does not have the same effect.  The other thing you could do is remove the trim() statements I added in the functional code, but keep the tests the same.  Then run the tests and see if they fail on MacOS.

Reply:
@skelliam Yes, I did exactly that and it works for me even before `trim()` was added.
I have added spaces after the last label and after each last value in each line, and then selected that label in plot configuration and specified it without trailing space. Plots rendered just fine.

Reply:
@vgaidarji, maybe it has something to do with the CR LF endings in Windows?  If you don't implement trim(), do the new tests pass?  If so that is really strange...

Reply:
I would really like to somehow have another user verify this -- I guess it is pretty low risk though.

Reply:
@skelliam yes, it's a very low risk. Trimming trailing spaces sounds very reasonable to me.
I've verified following cases:
1. plot plugin build from 1ccbcc8b6f5b3d21109e098c83c84f5a6baea058
- "column" & "value" & `include column by name` "column" = plots are shown
- "column " & "value " & `include column by name` "column" = plots are shown
- "column " & "value " & `include column by name` "column " = plots are **not** shown
- "column" & "value" & `include column by name` "column " = plots are **not** shown

2. Plot plugin 2.0.3 version
- "column" & "value" & `include column by name` "column" = plots are shown
- "column " & "value " & `include column by name` "column" = plots are shown
- "column " & "value " & `include column by name` "column " = plots are **not** shown
- "column" & "value" & `include column by name` "column " = plots are **not** shown

As you can see, the behavior is the same. I'm on Mac OS High Sierra 10.13.3.

---

I'm just wondering why I can't reproduce the issue you described and if it's OS related:
> If someone wants to reproduce this on Windows, I used this as a Windows batch 
```
set plotfile=test.csv

IF NOT EXIST %plotfile% (
   echo foo,bar,thing > %plotfile%
)

echo %time:~-5%,7,%time:~-2% >> %plotfile%
```
> Now try to plot only the last column named "thing" using the plot plugin. It will not get plotted because the plugin only discovers the name "thing " (with space).

As I'm not getting plots when column name has the trailing space.
> "column " & "value " & include column by name "column " = plots are not shown

@skelliam Could you please confirm if you have the same behavior?